### PR TITLE
#sFJutiak CRUD operations for routes

### PR DESCRIPTION
**Pull request by TuyizeeAnastase**
**What does this PR do?**
creating CRUD operations for Routes

**What tasks to be completed?**
-create Routes table in database
-create function controller 
-create APIs

**How can be manually tested?**
-clone this repo if you have not cloned it
-run `npm install`
-run `npm run dev`
-in postman send request `https://eagles-phant-ft-crud-op-c6f6cf.herokuapp.com/api/v1/routes `

**Relavant trello story**
https://trello.com/c/sFJutiak/7-crud-operations-for-routes

CRUS for routes

### CODE REVIEW

**Comment by Ally4** [comment]
work on the deployment

**Review by Ally4 — APPROVED** [review]

**Review by Ntwali-Josue — APPROVED** [review]

**Review by Mugema123 — APPROVED** [review]

**Comment by TuyizeeAnastase** [comment]
this PR is not deployed because we have got to the limit of deployed app on Heroku so wait one to be merged i will deploy it.

File: src/controllers/routes.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+const getOneRoute=async (req,res)=>{
+   try{
+       const { id }=req.params;
+       const route=await models.Routes.findOne({
+           where:{ id },
+       });
+       if(route){
+           return res.status(200).json({route});
+       }
+       return res.status(401).json({message:'No Route with that Id'});
```
Comment (ceelogre) [inline src/controllers/routes.js]: Message not translated.

File: src/controllers/routes.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    }
+};
+const deleteRoute=async(req,res)=>{
+    try{
+        const { id }=req.params;
+        const deleted=await models.Routes.destroy({
+            where:{ id }
+        });
+        if(deleted){
+            return res.status(204).send("Route deleted");
```
Comment (ceelogre) [inline src/controllers/routes.js]: Message is not properly formated.

File: src/controllers/routes.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    }
+}
+
+module.exports={
+    createRoutes,
+    getRoutes,
+    getOneRoute,
+    updateRoute,
+    deleteRoute
+}
```
Comment (ceelogre) [inline src/controllers/routes.js]: `module.exports` is discouraged in ES Modules.

File: src/test/registration.test.js [hunk 4]
```diff
@@ -74,9 +75,9 @@ describe('Tests for the registration of the users', () => {
       .set('x-access-token', adminToken)
       .send(user[2])
       .end((error, res) => {
-        expect(res).to.have.status([201]);
+        expect(res).to.have.status([403]);
         expect(res.body).to.have.property('status');
-        expect(res.body.status).to.be.equal(201);
+        expect(res.body.status).to.be.equal(403);
         expect(res.body).to.have.property('message');
         expect(res.body.message).to.be.a('string');
```
Comment (ceelogre) [inline src/test/registration.test.js:92]: Is there a possibility to assert the error message now that you know it's a string?

File: src/test/routes.test.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import app from '../app';
+
+
+chai.use(chaiHttp);
+const requester=()=>chai.request(app);
+const prefix='/api/v1/routes';
+
+const adminToken = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6IndpbGxAZ21haWwuY29tIiwicm9sZSI6ImFkbWluIiwiaWF0IjoxNjAzMzc3MDk5LCJleHAiOjE2MDM5ODE4OTl9.rEtLJMojuQPCLxyDTII1EXZQwvM--FK24lPzAmwHOsw';
+const fakeToken = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6Ikpvc2hAZ21haWwuY29tIiwicm9sZSI6ImFkbWluIiwiaWF0IjoxNjAyNzY0NDkwLCJleHAiOjE2MDMzNjkyOTB9.Jctf3hUMNigF19-ShwOg6_4p1gAwgD97E7PUHwdtNd';
+const nonAdminToken = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6ImJydWNlQGdtYWlsLmNvbSIsInJvbGUiOiJkcml2ZXIiLCJpYXQiOjE2MDMzNzcyNzAsImV4cCI6MTYwMzk4MjA3MH0.UCLiQvmKRhD8_iamHOnjMLgVtxs9Z1e1ixVgc0cVVUA';
```
Comment (ceelogre) [inline src/test/routes.test.js]: Variable seems unused.

**Review by ceelogre — CHANGES_REQUESTED** [review]
Giving an invalid id when patching a route seems to crash the server.
Capture d’écran 2020-12-23 à 15 56 30

**Review by ceelogre — APPROVED** [review]

**Pull request merged**